From f16e948ca2a826600e44ed2e694d9669b67810f9 Mon Sep 17 00:00:00 2001
From: Shahbaz Youssefi <syoussefi@chromium.org>
Date: Thu, 09 Jan 2020 14:20:09 -0500
Subject: [PATCH] Vulkan: Optimize getIndexRange

This function was unconditionally calling finishImpl().  This is changed
to finishToSerial() to only wait until the serial that actually used the
buffer.

Bug: angleproject:3072
Change-Id: Ida89bb119b4ba6420f12404b911af0e3b4583a51
---

diff --git a/src/libANGLE/renderer/vulkan/BufferVk.cpp b/src/libANGLE/renderer/vulkan/BufferVk.cpp
index 8ed1930..69fe052 100644
--- a/src/libANGLE/renderer/vulkan/BufferVk.cpp
+++ b/src/libANGLE/renderer/vulkan/BufferVk.cpp
@@ -296,7 +296,7 @@
 
     ANGLE_TRACE_EVENT0("gpu.angle", "BufferVk::getIndexRange");
     // Needed before reading buffer or we could get stale data.
-    ANGLE_TRY(contextVk->finishImpl());
+    ANGLE_TRY(contextVk->finishToSerial(mBuffer.getLatestSerial()));
 
     // TODO(jmadill): Consider keeping a shadow system memory copy in some cases.
     ASSERT(mBuffer.valid());
